test/python/transpiler/test_check_cnot_direction.py

```diff
@@ -0,0 +1,317 @@
+        coupling = CouplingMap([[0, 1], [1, 2], [2, 3]])
+        dag = circuit_to_dag(circuit)
+
+        pass_ = CheckCnotDirection(coupling)
+        pass_.run(dag)
+
+        self.assertFalse(pass_.property_set['is_direction_mapped'])
+
+
+class TestCheckMapCZ(QiskitTestCase):
+    """ Tests the CheckMap pass with CZ gates"""
+
+    def test_true_map(self):
+        """ Mapped is easy to check
+         qr0:--(Z)-[H]-(Z)-
+                |       |
+         qr1:---.-------|--
+                        |
+         qr2:-----------.--
+
+         CouplingMap map: [1]<-[0]->[2]
+        """
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.cz(qr[0], qr[1])
+        circuit.h(qr[0])
+        circuit.cz(qr[0], qr[2])
+        coupling = CouplingMap([[0, 1], [0, 2]])
+        dag = circuit_to_dag(circuit)
+
+        pass_ = CheckCnotDirection(coupling)
+        pass_.run(dag)
+
+        self.assertTrue(pass_.property_set['is_direction_mapped'])
+
+    def test_false_map(self):
+        """ Needs [0]-[1] in a [0]--[2]--[1]
+         qr0:--(Z)--
+                |
+         qr1:---.---
+
+         CouplingMap map: [0]->[2]->[1]
+        """
+        qr = QuantumRegister(2, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.cz(qr[0], qr[1])
+        coupling = CouplingMap([[0, 2], [2, 1]])
+        dag = circuit_to_dag(circuit)
+
+        pass_ = CheckCnotDirection(coupling)
+        pass_.run(dag)
+
+        self.assertFalse(pass_.property_set['is_direction_mapped'])
+
+    def test_true_map_undirected(self):
+        """ Mapped but with wrong direction
+         qr0:--(Z)-[H]--.--
+                |       |
+         qr1:---.-------|--
+                        |
+         qr2:----------(Z)-
+
+         CouplingMap map: [1]<-[0]->[2]
+        """
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.cz(qr[0], qr[1])
+        circuit.h(qr[0])
+        circuit.cz(qr[2], qr[0])
+        coupling = CouplingMap([[0, 1], [0, 2]])
+        dag = circuit_to_dag(circuit)
+
+        pass_ = CheckCnotDirection(coupling)
+        pass_.run(dag)
+
+        self.assertFalse(pass_.property_set['is_direction_mapped'])
+
+
+class TestCheckMapSwap(QiskitTestCase):
+    """ Tests the CheckMap pass with Swap gates"""
+
+    def test_true_map(self):
+        """ Mapped is easy to check
+
+         qr0:--X-[H]-X--
+               |     |
+         qr1:--X-----|--
+                     |
+         qr2:--------X--
+
+         CouplingMap map: [1]<-[0]->[2]
+        """
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.swap(qr[0], qr[1])
+        circuit.h(qr[0])
+        circuit.swap(qr[0], qr[2])
+        coupling = CouplingMap([[0, 1], [0, 2]])
+        dag = circuit_to_dag(circuit)
+
+        pass_ = CheckCnotDirection(coupling)
+        pass_.run(dag)
+
+        self.assertFalse(pass_.property_set['is_direction_mapped'])
```

Comment:
Should this be false? All the swaps are in the coupling map and the test it called `test_true_map` :)

Reply:
depending on https://github.com/Qiskit/qiskit-terra/pull/1767#discussion_r257090395

Reply:
test fixed in 501897b3 !